### PR TITLE
files: Don't create conflict for existing, but identical files

### LINKED ISSUE #1213: mimeapps.list get overriden and makes activation fail
Hi,

I've been trying out `xdg.mimeApps` and I love the idea. Overall, it works really well except some software override `~/.config/mimeapps.list` very frequently (I'm guessing Thunar) which not only lead to unwanted association but also make the next home-manager activation fail:
```
Hostname.......> <DATE> Hostname hm-activate-user[28434]: Activating home-manager configuration for user
Hostname.......> <DATE> Hostname hm-activate-user[28434]: Starting home manager activation
Hostname.......> <DATE> Hostname hm-activate-user[28434]: Activating checkFilesChanged
Hostname.......> <DATE> Hostname hm-activate-user[28434]: Activating checkLinkTargets
Hostname.......> <DATE> Hostname hm-activate-user[28434]: Existing file '/home/user/.config/mimeapps.list' is in the way
Hostname.......> <DATE> Hostname hm-activate-user[28434]: Please move the above files and try again or use -b <ext> to move automatically.
```

I've tried using an `home.activation` script in order to delete the `mimeapps.list` but unfortunately it seems those scripts are executed after home-manager tries to create the symlink.

Looking at past issues, I noticed I'm not the first person wondering (https://github.com/rycee/home-manager/pull/834#issuecomment-530527582) but I couldn't find a solution to this issue (apart from using the `-b` option every time, which is far from ideal).

Is there any way to either 
- Make the file read only (which as far as I know doesn't break anything) 
- Or run `rm ~/.config/mimeapps.list` **before** every home-manager activation
? 

Thank you for any help you might provide.

Unrelated question for @rycee (As I don't believe I should open an issue for this): Have you considered using Github sponsor or another type of donation for your work on `home-manager`? I'm sure some people would gladly try to support you financially for the time you spend on it.

### PR DESCRIPTION
**Pull request by piegamesde**
Fixes #1213

### Description

I tried implementing it in such a way that a new configuration option isn't needed for this feature. But I'm not 100% sure about it, there may be a few corner cases that I am missing.

### Checklist

- [x] Code formatted with `./format`.
- [x] Code tested through `nix-shell --pure tests -A run.all`.
- [ ] Test cases updated/added. See [example](https://github.com/rycee/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).
- [x] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

    See [CONTRIBUTING](https://github.com/rycee/home-manager/blob/master/CONTRIBUTING.md#commits) for more information and [recent commit messages](https://github.com/rycee/home-manager/commits/master) for examples.

### CODE REVIEW

**Comment by piegamesde** [comment]
I still stand by this pull request. In fact, I am using the patch for almost a full year now and it works very well. The patch still applies, but I can rebase it onto master nevertheless if requested.

**Comment by berbiche** [comment]
Hi @piegamesde,

Can you rebase your changes?

I'd be cool to have a unit test but I don't think it's possible to achieve with the way nmt scripts work.

**Comment by piegamesde** [comment]
Sorry for the noise. I just forgot that I hadn't used a feature branch for this one ^^. Should be rebased now, and the diff still looks sane.

Regarding unit test, if you point me the directions I might give it a try but no guarantees.

File: modules/files.nix [hunk 1]
```diff
@@ -169,11 +175,19 @@ in
             relativePath="''${sourcePath#$newGenFiles/}"
             targetPath="$HOME/$relativePath"
             if [[ -e "$targetPath" && ! -L "$targetPath" && -n "$HOME_MANAGER_BACKUP_EXT" ]] ; then
+              # The target exists, back it up
               backup="$targetPath.$HOME_MANAGER_BACKUP_EXT"
               $DRY_RUN_CMD mv $VERBOSE_ARG "$targetPath" "$backup" || errorEcho "Moving '$targetPath' failed!"
             fi
-            $DRY_RUN_CMD mkdir -p $VERBOSE_ARG "$(dirname "$targetPath")"
-            $DRY_RUN_CMD ln -nsf $VERBOSE_ARG "$sourcePath" "$targetPath"
+
```
Comment (rycee) [inline modules/files.nix:182]:
```suggestion

```

Comment (rycee) [reply modules/files.nix:182]: I'll fix this myself and merge to master…

**Comment by rycee** [comment]
Thanks! Rebased to master in 7f976da06840c268cc291a021bab7532b923713c.
